querybook/server/datasources/datadoc.py

```diff
@@ -367,6 +367,36 @@ def get_datadoc_editors(doc_id):
     return logic.get_data_doc_editors_by_doc_id(doc_id)
 
 
+@register("/datadoc/scheduled/", methods=["GET"])
+def get_datadoc_with_schedule(
+    environment_id,
+    filter_mode=None,
+    offset=0,
+    limit=10,
+    filtered_name=None,
+    archived=False,
+):
+    verify_environment_permission([environment_id])
+
+    docs = []
+
+    docs, count = logic.get_data_doc_by_user(
+        current_user.id,
+        environment_id=environment_id,
+        offset=offset,
+        limit=limit,
+        filtered_name=filtered_name,
+    )
+
+    scheduled_doc_names = [get_data_doc_schedule_name(doc.id) for doc in docs]
```

Comment:
Hey in case this comment wasn't clear. What I meant is in 

```
@with_session
def get_task_run_record_run_with_schedule(data_doc_names, docs, session):
```
you can do

```
@with_session
def get_task_run_record_run_with_schedule(docs, session):
    scheduled_doc_names = [get_data_doc_schedule_name(doc.id) for doc in docs]
```